app/server/src/features/accounts/methods.unit.test.ts

```diff
@@ -0,0 +1,418 @@
+            prismaMock.user.findUnique.mockResolvedValueOnce(user);
+
+            // Assert
+            await expect(AccountMethods.loginWithPassword(prismaMock, input)).rejects.toThrow(
+                'Login failed; Invalid email or password.'
+            );
+        });
+
+        test('should login user successfully', async () => {
+            // Arrange
+            const user = { ...userData };
+            const input = { email: userData.email, password: userTextPassword };
+            prismaMock.user.findUnique.mockResolvedValueOnce(user);
+
+            // Act
+            const output = await AccountMethods.loginWithPassword(prismaMock, input);
+
+            // Assert
+            const userWithGlobalId = toUserId(user);
+            const token = await jwt.sign({ id: userWithGlobalId.id });
+            const expectedOutput = { user, token };
+            expect(output).toEqual(expectedOutput);
+        });
+    });
+
+    describe('updateEmail', () => {
+        test('should throw if no user is account with new email already exists', async () => {
+            // Arrange
+            const user = { ...userData };
+            const input = { currentEmail: 'test@test.com', newEmail: userData.email };
+            prismaMock.user.findUnique.mockResolvedValueOnce(user);
+
+            // Assert
+            await expect(AccountMethods.updateEmail(prismaMock, input)).rejects.toThrow(
+                'Updating email failed: Another user exists with this email. Please input a different email.'
+            );
+        });
+        test('should return updated user and token when successful', async () => {
+            // Arrange
+            const newEmail = 'newEmail@test.com';
+            const user = { ...userData, email: newEmail };
+            const input = { currentEmail: userData.email, newEmail };
+            prismaMock.user.update.mockResolvedValueOnce(user);
+
+            // Act
+            const output = await AccountMethods.updateEmail(prismaMock, input);
+
+            // Assert
+            const userWithGlobalId = toUserId(user);
+            const token = await jwt.sign({ id: userWithGlobalId.id });
+            const expectedOutput = { updatedUser: user, token };
+            expect(output).toEqual(expectedOutput);
+        });
+    });
+
+    describe('updatePassword', () => {
+        test('should throw if account does not exist', async () => {
+            // Arrange
+            const newPassword = 'newPassword1!';
+            const input = {
+                email: 'dne@test.com',
+                oldPassword: userTextPassword,
+                newPassword,
+                confirmNewPassword: newPassword,
+            };
+            prismaMock.user.findUnique.mockResolvedValueOnce(null);
+
+            // Assert
+            await expect(AccountMethods.updatePassword(prismaMock, input)).rejects.toThrow('Account not found.');
```

Comment:
We'll need to change these messages to follow owasp recommendations (I've mentioned these before).

Reply:
Yeah I wanted to do that but I thought it's best to keep that as a separate PR and not to mix it with this one.